include/deal.II/base/mpi.h

```diff
@@ -534,6 +534,27 @@ namespace Utilities
     all_gather(const MPI_Comm &comm,
                const T        &object_to_send);
 
+    /**
+     * A generalization of the classic MPI_Gather function, that accepts
+     * arbitrary data types T, as long as boost::serialize accepts T as an
+     * argument.
+     *
+     * @param[in] comm MPI communicator.
+     * @param[in] object_to_send An object to send to the root process
+     *
+     * @return A vector of objects, with size equal to the number of
+     *  processes in the MPI communicator. Each entry contains the object
+     *  received from the processor with the corresponding rank within the
+     *  communicator.
```

Comment:
Can you document that this non-empty vector is really only returned on the *root* process, whereas on all other processors, it is an empty vector?